### ISSUE TITLE
Explicit listAudioOutputDevices() call is required to call to chooseAudioOutputDevice(..)

### ISSUE THREAD

**Issue by malsabbagh** [aws/amazon-chime-sdk-js#1245]
**Describe the bug**
It seems the logic in DefaultDeviceController.ts for [`chooseAudioOutputDevice(..)`](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/DefaultDeviceController.ts#L540) doesn't attempt to update `deviceInfoCache` when it is null. However, this is not the same when dealing with [`chooseAudioInputDevice()`](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/DefaultDeviceController.ts#L289) or [`chooseVideoInputDevice()`](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/DefaultDeviceController.ts#L515) as both call [`chooseInputIntrinsicDevice()`](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/DefaultDeviceController.ts#L1301) which eventually calls [`navigator.mediaDevices.enumerateDevices()`](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/DefaultDeviceController.ts#L1031)

Is there a reason why the behaviour is different between AudioOutput and both AudioInput and VideoInput?

**Platform**
 - OS: Mac
 - Browser: Chrome
 - Browser Version: Latest

**Comment by ltrung** [comment]
@malsabbagh Could you point to the code where `chooseInputIntrinsicDevice` calls `enumurateDevices`? As far as I know, both `chooseAudioInputDevice` and `chooseVideoInputDevice` need to call list devices first. You can see that the `matchesDeviceSelection` call in `chooseInputIntrinsicDevice` assume device cache is populated already.

**Comment by malsabbagh** [comment]
@ltrung ah, I guess the path I tried to reference is wrong. Thank you for pointing that out. Moreover, It seems `deviceInfoCache` was not the reason why both `chooseAudioInputDevice()` and `chooseVideoInputDevice()` were working. The path for that is a bit more complicated.

I did a bit more testing and I can confirm that the behavior I was describing was specific to `amazon-chime-sdk-js` up to `2.5.0`

After that version, if I attempt to call `chooseAudioInputDevice()` without calling the corresponding `listAudioInputDevices()` I get into some infinite loop. 

Given it is a requirement to call `list` before calling `choose`, it might be good to clarify that in the documentation or maybe to even throw an exception when an attempt to call `choose` was made without `list`
